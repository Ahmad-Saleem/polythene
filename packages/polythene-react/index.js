export * from "polythene-react-base";
export * from "polythene-react-button";
export * from "polythene-react-card";
export * from "polythene-react-checkbox";
export * from "polythene-react-dialog";
export * from "polythene-react-dialog-pane";
export * from "polythene-react-fab";
export * from "polythene-react-icon";
export * from "polythene-react-icon-button";
export * from "polythene-react-list";
export * from "polythene-react-list-tile";
export * from "polythene-react-menu";
export * from "polythene-react-radio-button";
export * from "polythene-react-raised-button";
export * from "polythene-react-ripple";
export * from "polythene-react-shadow";
export * from "polythene-react-svg";
export * from "polythene-react-switch";
export * from "polythene-react-textfield";
export * from "polythene-react-toolbar";
