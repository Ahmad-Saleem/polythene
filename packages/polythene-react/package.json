{
  "name": "polythene-react",
  "version": "1.0.0-rc.8",
  "description": "",
  "main": "dist/polythene-react.js",
  "module": "dist/polythene-react.mjs",
  "scripts": {
    "clean": "rm -rf dist && mkdir dist",
    "rollup": "../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.umd.js && ../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.es.js",
    "build-standalone": "DEPS=1 ENTRY=scripts/standalone.js SOURCEMAP=1 DEST=./dist/polythene-react-standalone.js ../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.umd-includes.js",
    "build": "yarn run rollup && yarn run build-standalone"
  },
  "files": [
    "dist",
    "README.md"
  ],
  "dependencies": {
    "polythene-css": "^1.0.0-rc.8",
    "polythene-react-base": "^1.0.0-rc.8",
    "polythene-react-button": "^1.0.0-rc.8",
    "polythene-react-card": "^1.0.0-rc.8",
    "polythene-react-checkbox": "^1.0.0-rc.8",
    "polythene-react-dialog": "^1.0.0-rc.8",
    "polythene-react-dialog-pane": "^1.0.0-rc.8",
    "polythene-react-fab": "^1.0.0-rc.8",
    "polythene-react-icon": "^1.0.0-rc.8",
    "polythene-react-icon-button": "^1.0.0-rc.8",
    "polythene-react-ios-spinner": "^1.0.0-rc.8",
    "polythene-react-list": "^1.0.0-rc.8",
    "polythene-react-list-tile": "^1.0.0-rc.8",
    "polythene-react-material-design-progress-spinner": "^1.0.0-rc.8",
    "polythene-react-material-design-spinner": "^1.0.0-rc.8",
    "polythene-react-menu": "^1.0.0-rc.8",
    "polythene-react-notification": "^1.0.0-rc.8",
    "polythene-react-radio-button": "^1.0.0-rc.8",
    "polythene-react-radio-group": "^1.0.0-rc.8",
    "polythene-react-raised-button": "^1.0.0-rc.8",
    "polythene-react-ripple": "^1.0.0-rc.8",
    "polythene-react-search": "^1.0.0-rc.8",
    "polythene-react-shadow": "^1.0.0-rc.8",
    "polythene-react-slider": "^1.0.0-rc.8",
    "polythene-react-snackbar": "^1.0.0-rc.8",
    "polythene-react-svg": "^1.0.0-rc.8",
    "polythene-react-switch": "^1.0.0-rc.8",
    "polythene-react-tabs": "^1.0.0-rc.8",
    "polythene-react-textfield": "^1.0.0-rc.8",
    "polythene-react-toolbar": "^1.0.0-rc.8"
  },
  "devDependencies": {
    "polythene-core": "^1.0.0-rc.8",
    "polythene-core-css": "^1.0.0-rc.8",
    "polythene-fastclick": "^1.0.0-rc.8",
    "react": "^15.6.1",
    "react-hyperscript": "^3.0.0"
  },
  "author": "Arthur Clemens <arthurclemens@gmail.com> (http://arthurclemens.com)",
  "license": "MIT"
}
