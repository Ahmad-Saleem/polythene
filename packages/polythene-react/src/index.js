export * from "./core";
export { button } from "./button";
export { icon } from "./icon";
export { ripple } from "./ripple";
export { shadow } from "./shadow";
export { svg } from "./svg";
