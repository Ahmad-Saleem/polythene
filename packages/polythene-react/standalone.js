export * from "./index";
import { addLayoutStyles } from "polythene-utilities";
import { addTypography, addRoboto } from "polythene-style";
import { addFastClick } from "polythene-fastclick";

addTypography();
addRoboto();
addLayoutStyles();
addFastClick();
