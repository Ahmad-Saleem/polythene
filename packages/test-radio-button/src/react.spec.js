import { runSnapshots } from "../../../scripts/tests/react-snapshots";
import { renderer, keys, RadioButton, RadioGroup } from "polythene-react";
import specTestsRadioButton from "./spec-tests-radio-button.js";
import specTestsRadioGroup from "./spec-tests-radio-group.js";
import reactTests from "./tests-react.js";

runSnapshots({
  tests: specTestsRadioButton({ RadioButton, renderer, keys }).concat(specTestsRadioGroup({ RadioGroup, renderer, keys })).concat(reactTests),
  renderer
});
