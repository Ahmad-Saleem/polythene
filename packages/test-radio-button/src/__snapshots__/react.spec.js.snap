exports[`test Dark tone class + light theme class 1`] = `
<div
  className="pe-light-tone"
  style={
    Object {
      "background": "#fff",
      "padding": "20px",
    }
  }>
  <RadioGroup
    content={
      Array [
        Object {
          "label": "One",
          "value": "One",
        },
        Object {
          "defaultChecked": true,
          "label": "Two",
          "value": "Two",
        },
      ]
    }
    name="test-dark-tone-light-class" />
</div>
`;

exports[`test Dark tone class + light tone 1`] = `
<div
  style={
    Object {
      "background": "#fff",
      "padding": "20px",
    }
  }>
  <RadioGroup
    content={
      Array [
        Object {
          "label": "One",
          "value": "One",
        },
        Object {
          "defaultChecked": true,
          "label": "Two",
          "value": "Two",
        },
      ]
    }
    name="test-dark-tone-light" />
</div>
`;

exports[`test No options - RadioButton 1`] = `
<SelectionControl
  icons={
    Object {
      "iconOff": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\"/></svg>",
      "iconOn": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\"/></svg>",
    }
  }
  instanceClass="pe-radio-control"
  selectable={[Function]}
  theme={[Function]}
  type="radio"
  value="1" />
`;

exports[`test No options - RadioGroup 1`] = `
<div
  className="pe-radio-group   " />
`;

exports[`test Option: Radio Group onChange 1`] = `
<div>
  <div
    style={
      Object {
        "margin": "0 0 1rem 0",
      }
    }>
    Value: Not set
  </div>
  <RadioGroup
    content={
      Array [
        Object {
          "label": "One",
          "value": "One",
        },
        Object {
          "label": "Two",
          "value": "Two",
        },
      ]
    }
    name="onChange"
    onChange={[Function]} />
</div>
`;

exports[`test Option: after - RadioGroup 1`] = `
<div
  className="pe-radio-group   ">
  <RadioButton
    checked={false}
    label="One"
    onChange={[Function]}
    value="One" />
  <RadioButton
    checked={false}
    label="Two"
    onChange={[Function]}
    value="Two" />
  <div>
    After
  </div>
</div>
`;

exports[`test Option: before - RadioGroup 1`] = `
<div
  className="pe-radio-group   ">
  <RadioButton
    checked={false}
    label="One"
    onChange={[Function]}
    value="One" />
  <RadioButton
    checked={false}
    label="Two"
    onChange={[Function]}
    value="Two" />
  <div>
    Before
  </div>
</div>
`;

exports[`test Option: buttons - RadioGroup 1`] = `
<div
  className="pe-radio-group   ">
  <RadioButton
    checked={false}
    label="One"
    name="label"
    onChange={[Function]}
    value="One" />
  <RadioButton
    checked={false}
    label="Two"
    name="label"
    onChange={[Function]}
    value="Two" />
</div>
`;

exports[`test Option: class - RadioButton 1`] = `
<SelectionControl
  className="class-x"
  icons={
    Object {
      "iconOff": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\"/></svg>",
      "iconOn": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\"/></svg>",
    }
  }
  instanceClass="pe-radio-control"
  selectable={[Function]}
  theme={[Function]}
  type="radio"
  value="1" />
`;

exports[`test Option: class - RadioGroup 1`] = `
<div
  className="pe-radio-group   class-x">
  <RadioButton
    checked={false}
    label="One"
    onChange={[Function]}
    value="One" />
  <RadioButton
    checked={false}
    label="Two"
    onChange={[Function]}
    value="Two" />
</div>
`;

exports[`test Option: className - RadioButton 1`] = `
<SelectionControl
  className="className-x"
  icons={
    Object {
      "iconOff": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\"/></svg>",
      "iconOn": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\"/></svg>",
    }
  }
  instanceClass="pe-radio-control"
  selectable={[Function]}
  theme={[Function]}
  type="radio"
  value="1" />
`;

exports[`test Option: className - RadioGroup 1`] = `
<div
  className="pe-radio-group   className-x">
  <RadioButton
    checked={false}
    label="One"
    onChange={[Function]}
    value="One" />
  <RadioButton
    checked={false}
    label="Two"
    onChange={[Function]}
    value="Two" />
</div>
`;

exports[`test Option: defaultChecked 1`] = `
<div
  className="pe-radio-group   ">
  <RadioButton
    checked={false}
    label="One"
    name="defaultChecked"
    onChange={[Function]}
    value="One" />
  <RadioButton
    checked={true}
    defaultChecked={true}
    label="Two"
    name="defaultChecked"
    onChange={[Function]}
    value="Two" />
</div>
`;

exports[`test Option: disabled -- dark tone 1`] = `
<div
  className="pe-radio-group   ">
  <RadioButton
    checked={false}
    disabled={true}
    label="One"
    name="disabled-dark"
    onChange={[Function]}
    value="One" />
  <RadioButton
    checked={true}
    disabled={true}
    label="Two"
    name="disabled-dark"
    onChange={[Function]}
    value="Two" />
</div>
`;

exports[`test Option: disabled 1`] = `
<div
  className="pe-radio-group   ">
  <RadioButton
    checked={false}
    disabled={true}
    label="One"
    name="disabled"
    onChange={[Function]}
    value="One" />
  <RadioButton
    checked={true}
    disabled={true}
    label="Two"
    name="disabled"
    onChange={[Function]}
    value="Two" />
</div>
`;

exports[`test Option: element - RadioButton 1`] = `
<SelectionControl
  element="a"
  icons={
    Object {
      "iconOff": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\"/></svg>",
      "iconOn": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\"/></svg>",
    }
  }
  instanceClass="pe-radio-control"
  selectable={[Function]}
  theme={[Function]}
  type="radio"
  value="1" />
`;

exports[`test Option: element - RadioGroup 1`] = `
<a
  className="pe-radio-group   ">
  <RadioButton
    checked={false}
    label="One"
    onChange={[Function]}
    value="One" />
  <RadioButton
    checked={false}
    label="Two"
    onChange={[Function]}
    value="Two" />
</a>
`;

exports[`test Option: iconButton (custom hover behaviour) 1`] = `
<div
  className="pe-radio-group   ">
  <RadioButton
    checked={false}
    iconButton={
      Object {
        "ink": false,
        "wash": true,
      }
    }
    label="Hover me"
    name="iconButton"
    onChange={[Function]}
    value="One" />
  <RadioButton
    checked={false}
    iconButton={
      Object {
        "ink": false,
        "wash": true,
      }
    }
    label="Hover me"
    name="iconButton"
    onChange={[Function]}
    value="Two" />
</div>
`;

exports[`test Option: iconOn, iconOff (custom icon) 1`] = `
<div
  className="pe-radio-group   "
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }>
  <RadioButton
    checked={false}
    iconOff={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24.00 24.00\" enable-background=\"new 0 0 24.00 24.00\" xml:space=\"preserve\"><path fill=\"#000000\" fill-opacity=\"1\" stroke-width=\"0.2\" stroke-linejoin=\"round\" d=\"M 11.9994,15.3943L 8.2364,17.6643L 9.2314,13.3833L 5.9094,10.5053L 10.2894,10.1293L 11.9994,6.09327L 13.7094,10.1293L 18.0894,10.5053L 14.7674,13.3833L 15.7624,17.6643M 21.9994,9.24227L 14.8084,8.62526L 11.9994,1.99827L 9.1904,8.62526L 1.9994,9.24227L 7.4544,13.9693L 5.8194,20.9983L 11.9994,17.2703L 18.1794,20.9983L 16.5444,13.9693L 21.9994,9.24227 Z \"/></svg>",
                }
          } />,
      }
    }
    iconOn={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\"/></svg>",
                }
          } />,
      }
    }
    name="icon"
    onChange={[Function]}
    size="small"
    value="small" />
  <RadioButton
    checked={false}
    iconOff={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24.00 24.00\" enable-background=\"new 0 0 24.00 24.00\" xml:space=\"preserve\"><path fill=\"#000000\" fill-opacity=\"1\" stroke-width=\"0.2\" stroke-linejoin=\"round\" d=\"M 11.9994,15.3943L 8.2364,17.6643L 9.2314,13.3833L 5.9094,10.5053L 10.2894,10.1293L 11.9994,6.09327L 13.7094,10.1293L 18.0894,10.5053L 14.7674,13.3833L 15.7624,17.6643M 21.9994,9.24227L 14.8084,8.62526L 11.9994,1.99827L 9.1904,8.62526L 1.9994,9.24227L 7.4544,13.9693L 5.8194,20.9983L 11.9994,17.2703L 18.1794,20.9983L 16.5444,13.9693L 21.9994,9.24227 Z \"/></svg>",
                }
          } />,
      }
    }
    iconOn={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\"/></svg>",
                }
          } />,
      }
    }
    name="icon"
    onChange={[Function]}
    size="regular"
    value="regular" />
  <RadioButton
    checked={false}
    iconOff={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24.00 24.00\" enable-background=\"new 0 0 24.00 24.00\" xml:space=\"preserve\"><path fill=\"#000000\" fill-opacity=\"1\" stroke-width=\"0.2\" stroke-linejoin=\"round\" d=\"M 11.9994,15.3943L 8.2364,17.6643L 9.2314,13.3833L 5.9094,10.5053L 10.2894,10.1293L 11.9994,6.09327L 13.7094,10.1293L 18.0894,10.5053L 14.7674,13.3833L 15.7624,17.6643M 21.9994,9.24227L 14.8084,8.62526L 11.9994,1.99827L 9.1904,8.62526L 1.9994,9.24227L 7.4544,13.9693L 5.8194,20.9983L 11.9994,17.2703L 18.1794,20.9983L 16.5444,13.9693L 21.9994,9.24227 Z \"/></svg>",
                }
          } />,
      }
    }
    iconOn={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\"/></svg>",
                }
          } />,
      }
    }
    name="icon"
    onChange={[Function]}
    size="medium"
    value="medium" />
  <RadioButton
    checked={false}
    iconOff={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24.00 24.00\" enable-background=\"new 0 0 24.00 24.00\" xml:space=\"preserve\"><path fill=\"#000000\" fill-opacity=\"1\" stroke-width=\"0.2\" stroke-linejoin=\"round\" d=\"M 11.9994,15.3943L 8.2364,17.6643L 9.2314,13.3833L 5.9094,10.5053L 10.2894,10.1293L 11.9994,6.09327L 13.7094,10.1293L 18.0894,10.5053L 14.7674,13.3833L 15.7624,17.6643M 21.9994,9.24227L 14.8084,8.62526L 11.9994,1.99827L 9.1904,8.62526L 1.9994,9.24227L 7.4544,13.9693L 5.8194,20.9983L 11.9994,17.2703L 18.1794,20.9983L 16.5444,13.9693L 21.9994,9.24227 Z \"/></svg>",
                }
          } />,
      }
    }
    iconOn={
      Object {
        "svg": <div
          dangerouslySetInnerHTML={
              Object {
                  "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\"/></svg>",
                }
          } />,
      }
    }
    name="icon"
    onChange={[Function]}
    size="large"
    value="large" />
</div>
`;

exports[`test Option: id - RadioButton 1`] = `
<SelectionControl
  icons={
    Object {
      "iconOff": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\"/></svg>",
      "iconOn": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\"/></svg>",
    }
  }
  id="id-x"
  instanceClass="pe-radio-control"
  selectable={[Function]}
  theme={[Function]}
  type="radio"
  value="1" />
`;

exports[`test Option: id - RadioGroup 1`] = `
<div
  className="pe-radio-group   "
  id="id-x">
  <RadioButton
    checked={false}
    label="One"
    onChange={[Function]}
    value="One" />
  <RadioButton
    checked={false}
    label="Two"
    onChange={[Function]}
    value="Two" />
</div>
`;

exports[`test Option: label (JSX) 1`] = `
<RadioGroup
  buttons={
    Array [
      Object {
        "label": "One",
        "value": "One",
      },
      Object {
        "label": "Two",
        "value": "Two",
      },
    ]
  } />
`;

exports[`test Option: label -- dark tone 1`] = `
<div
  className="pe-radio-group   ">
  <RadioButton
    checked={false}
    label="One"
    name="label-dark"
    onChange={[Function]}
    value="One" />
  <RadioButton
    checked={false}
    label="Two"
    name="label-dark"
    onChange={[Function]}
    value="Two" />
</div>
`;

exports[`test Option: label 1`] = `
<div
  className="pe-radio-group   ">
  <RadioButton
    checked={false}
    label="One"
    name="label"
    onChange={[Function]}
    value="One" />
  <RadioButton
    checked={false}
    label="Two"
    name="label"
    onChange={[Function]}
    value="Two" />
</div>
`;

exports[`test Option: onChange (JSX) 1`] = `
<form>
  <div
    style={
      Object {
        "margin": "0 0 1rem 0",
      }
    }>
    Value: undefined
  </div>
  <RadioButton
    checked={false}
    label="One"
    onChange={[Function]}
    value="one" />
  <RadioButton
    checked={false}
    label="Two"
    onChange={[Function]}
    value="two" />
</form>
`;

exports[`test Option: selectable 1`] = `
<div
  className="pe-radio-group   ">
  <RadioButton
    checked={false}
    label="Never"
    name="selectable"
    onChange={[Function]}
    selectable={[Function]}
    value="One" />
  <RadioButton
    checked={false}
    label="Only when unchecked"
    name="selectable"
    onChange={[Function]}
    selectable={[Function]}
    value="Two" />
</div>
`;

exports[`test Option: size 1`] = `
<div
  className="pe-radio-group   "
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }>
  <RadioButton
    checked={false}
    label="small"
    name="size"
    onChange={[Function]}
    size="small"
    value="small" />
  <RadioButton
    checked={false}
    label="regular"
    name="size"
    onChange={[Function]}
    size="regular"
    value="regular" />
  <RadioButton
    checked={false}
    label="medium"
    name="size"
    onChange={[Function]}
    size="medium"
    value="medium" />
  <RadioButton
    checked={false}
    label="large"
    name="size"
    onChange={[Function]}
    size="large"
    value="large" />
</div>
`;

exports[`test Option: style (colors) 1`] = `
<div
  className="pe-radio-group   ">
  <RadioButton
    checked={false}
    label="One"
    name="style"
    onChange={[Function]}
    style={
      Object {
        "color": "#EF6C00",
      }
    }
    value="One" />
  <RadioButton
    checked={true}
    defaultChecked={true}
    label="Two"
    name="style"
    onChange={[Function]}
    style={
      Object {
        "color": "#EF6C00",
      }
    }
    value="Two" />
</div>
`;

exports[`test Option: style - RadioGroup 1`] = `
<div
  className="pe-radio-group   "
  style={
    Object {
      "buttons": Array [
        Object {
          "label": "One",
          "value": "One",
        },
        Object {
          "label": "Two",
          "value": "Two",
        },
      ],
      "color": "red",
    }
  } />
`;

exports[`test Option: tabindex - RadioButton 1`] = `
<SelectionControl
  icons={
    Object {
      "iconOff": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\"/></svg>",
      "iconOn": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\"/></svg>",
    }
  }
  instanceClass="pe-radio-control"
  selectable={[Function]}
  tabIndex={3}
  theme={[Function]}
  type="radio"
  value="1" />
`;

exports[`test Option: value - RadioButton 1`] = `
<SelectionControl
  icons={
    Object {
      "iconOff": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\"/></svg>",
      "iconOn": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\"/></svg>",
    }
  }
  instanceClass="pe-radio-control"
  name="worth"
  selectable={[Function]}
  theme={[Function]}
  type="radio"
  value="notable" />
`;

exports[`test Themed radio button (color and font size) -- dark tone 1`] = `
<div
  className="pe-radio-group   ">
  <RadioButton
    checked={false}
    className="tests-radio-button-themed-radio"
    label="One"
    name="themed-dark"
    onChange={[Function]}
    value="One" />
  <RadioButton
    checked={true}
    className="tests-radio-button-themed-radio"
    defaultChecked={true}
    label="Two"
    name="themed-dark"
    onChange={[Function]}
    value="Two" />
</div>
`;

exports[`test Themed radio button (color and font size) 1`] = `
<div
  className="pe-radio-group   ">
  <RadioButton
    checked={false}
    className="tests-radio-button-themed-radio"
    label="One"
    name="themed"
    onChange={[Function]}
    value="One" />
  <RadioButton
    checked={true}
    className="tests-radio-button-themed-radio"
    defaultChecked={true}
    label="Two"
    name="themed"
    onChange={[Function]}
    value="Two" />
</div>
`;
